controller/iteration_blackbox_test.go

```diff
@@ -1269,4 +1269,73 @@ func (rest *TestIterationREST) TestUpdateIteration() {
 		svc, ctrl := rest.SecuredControllerWithIdentity(fxt.Identities[0])
 		test.UpdateIterationBadRequest(t, svc.Context, svc, ctrl, itr1.ID.String(), &payload)
 	})
+
+	rest.T().Run("update fail - parent UUID is same as subject iteraton", func(t *testing.T) {
+		fxt := tf.NewTestFixture(t, rest.DB, tf.CreateWorkItemEnvironment(), tf.Iterations(2))
+		typeIterationString := iteration.APIStringTypeIteration
+		itr1 := fxt.Iterations[1]
+		newParentIDStr := itr1.ID.String()
+		payload := app.UpdateIterationPayload{
+			Data: &app.Iteration{
+				Attributes: &app.IterationAttributes{},
+				Relationships: &app.IterationRelations{
+					Parent: &app.RelationGeneric{
+						Data: &app.GenericData{
+							ID:   &newParentIDStr,
+							Type: &typeIterationString,
+						},
+					},
+				},
+				ID:   &itr1.ID,
+				Type: typeIterationString,
+			},
+		}
+		svc, ctrl := rest.SecuredControllerWithIdentity(fxt.Identities[0])
+		test.UpdateIterationForbidden(t, svc.Context, svc, ctrl, itr1.ID.String(), &payload)
+	})
+
+	rest.T().Run("update fail - valid parent but from different space", func(t *testing.T) {
+		fxt := tf.NewTestFixture(t, rest.DB,
```

Comment:
Two be honest, I'd just created two test fixtures like so:

```go
fxt1 := tf.NewTestFixture(t, rest.DB, tf.Iterations(1, tf.SetIterationNames("alpha"))
fxt2 := tf.NewTestFixture(t, rest.DB, tf.Iterations(1, tf.SetIterationNames("beta"))

payload := app.UpdateIterationPayload{
  Data: ConvertIteration(&http.Request{Host: "api.service.domain.org"}, iteration.Iteration{
    ID: fxt2.IterationByName("beta").ID,
    Path: path.Path{fxt1.IterationByName("alpha").ID, fxt2.IterationByName("beta").ID},
  }),
}
svc, ctrl := rest.SecuredControllerWithIdentity(fxt1.Identities[0])
test.UpdateIterationForbidden(t, svc.Context, svc, ctrl, fxt2.IterationByName("beta").ID.String(), &payload)
```
Then `alpha` and `beta` automatically` belong to a different space. I don't see a need to create more than one iteration per space and I find my setup quite more readable than using a customization function.

Reply:
true that is simpler than previous.
done in https://github.com/fabric8-services/fabric8-wit/pull/1770/commits/75622623f469327dd41526bd841df5dae623e6a2